### PR TITLE
[SPARK-24743] [Examples]Update the JavaDirectKafkaWordCount example to support the new API of kafka

### PR DESCRIPTION
**Pull request by cluo512**
## What changes were proposed in this pull request?

Add some required configs for Kafka consumer in JavaDirectKafkaWordCount class.

## How was this patch tested?

Manual tests on Local mode.

### CODE REVIEW

**Comment by cluo512** [comment]
@srowen

**Comment by HyukjinKwon** [comment]
I would fill the PR description to explain what changes were proposed in more details. cc @koeninger

File: examples/src/main/java/org/apache/spark/examples/streaming/JavaDirectKafkaWordCount.java [hunk 1]
```diff
@@ -26,8 +26,10 @@
 
 import scala.Tuple2;
 
+import org.apache.kafka.clients.consumer.ConsumerConfig;
 import org.apache.kafka.clients.consumer.ConsumerRecord;
 
+import org.apache.kafka.common.serialization.StringDeserializer;
```
Comment (viirya) [inline examples/src/main/java/org/apache/spark/examples/streaming/JavaDirectKafkaWordCount.java]: I think this import should be moved to above?

**Comment by viirya** [comment]
This seems to be on par with Scala example `DirectKafkaWordCount`.

File: examples/src/main/java/org/apache/spark/examples/streaming/JavaDirectKafkaWordCount.java [hunk 2]
```diff
@@ -50,25 +52,30 @@
   private static final Pattern SPACE = Pattern.compile(" ");
 
   public static void main(String[] args) throws Exception {
-    if (args.length < 2) {
-      System.err.println("Usage: JavaDirectKafkaWordCount <brokers> <topics>\n" +
-          "  <brokers> is a list of one or more Kafka brokers\n" +
-          "  <topics> is a list of one or more kafka topics to consume from\n\n");
+    if (args.length < 3) {
+      System.err.println("Usage: JavaDirectKafkaWordCount <brokers> <groupId > <topics>\n" +
```
Comment (viirya) [inline examples/src/main/java/org/apache/spark/examples/streaming/JavaDirectKafkaWordCount.java]: `<groupId >` -> `<groupId>`.

File: examples/src/main/java/org/apache/spark/examples/streaming/JavaDirectKafkaWordCount.java [hunk 3]
```diff
@@ -50,25 +52,30 @@
   private static final Pattern SPACE = Pattern.compile(" ");
 
   public static void main(String[] args) throws Exception {
```
Comment (viirya) [inline examples/src/main/java/org/apache/spark/examples/streaming/JavaDirectKafkaWordCount.java]: We should also update the document in the code comment at the beginning of `JavaDirectKafkaWordCount`.

Comment (cluo512) [reply examples/src/main/java/org/apache/spark/examples/streaming/JavaDirectKafkaWordCount.java]: viirya, thank you for your nice review.

**Review by srowen — APPROVED** [review]

**Review by koeninger — APPROVED** [review]
LGTM, merging to master

**Pull request closed without merging**